app/controllers/admin_controller.rb

```diff
@@ -39,119 +130,110 @@ def create_user
-    raise "JURISDICTION must be provided and one of #{jurisdictions}" unless jurisdiction && jurisdictions.include?(jurisdiction)
+    return head :bad_request unless jurisdiction && jurisdictions.include?(jurisdiction)
 
-    user = User.find_by(email: email)
-    cur_jur = current_user.jurisdiction
-    redirect_to(root_url) && return unless (cur_jur.descendant_ids + [cur_jur.id]).include? user.jurisdiction.id
-    raise 'USER not found' unless user
+    is_api_enabled = permitted_params[:is_api_enabled]
+    return head :bad_request unless [true, false].include? is_api_enabled
 
-    user.jurisdiction = Jurisdiction.find_by_id(jurisdiction)
-    user.roles = []
-    user.add_role role.to_sym
-    user.save!
-  end
+    is_locked = permitted_params[:is_locked]
+    return head :bad_request unless [true, false].include? is_locked
 
-  def lock_user
-    redirect_to(root_url) && return unless current_user.has_role? :admin
+    # Find user
+    user = User.find_by(id: id)
+    return head :bad_request unless user
 
-    permitted_params = params[:admin].permit(:email)
-    email = permitted_params[:email]
-    user = User.find_by(email: email)
+    # Verify user jurisdiction access
     cur_jur = current_user.jurisdiction
     redirect_to(root_url) && return unless (cur_jur.descendant_ids + [cur_jur.id]).include? user.jurisdiction.id
 
-    user.lock_access!
-  end
+    # Update email
+    user.email = email
 
-  def reset_2fa
-    redirect_to(root_url) && return unless current_user.has_role? :admin
+    # Update jurisdiction
+    user.jurisdiction = Jurisdiction.find_by_id(jurisdiction)
 
-    permitted_params = params[:admin].permit(:email)
-    email = permitted_params[:email]
-    user = User.find_by(email: email)
-    cur_jur = current_user.jurisdiction
-    redirect_to(root_url) && return unless (cur_jur.descendant_ids + [cur_jur.id]).include? user.jurisdiction.id
+    # Update role
+    user.roles = []
+    user.add_role role.to_sym
+
+    # Update API access
+    user.update!(api_enabled: is_api_enabled)
+
+    # Update locked status
+    if user.locked_at.nil? && is_locked
+      user.lock_access!
+    elsif !user.locked_at.nil? && !is_locked
+      user.unlock_access!
+    end
 
-    user.authy_id = nil
-    user.authy_enabled = false
     user.save!
   end
 
-  def unlock_user
+  def reset_2fa
     redirect_to(root_url) && return unless current_user.has_role? :admin
 
-    permitted_params = params[:admin].permit(:email)
-    email = permitted_params[:email]
-    user = User.find_by(email: email)
+    permitted_params = params[:admin].permit({ ids: [] })
+    ids = permitted_params[:ids]
+    users = User.where(id: ids)
     cur_jur = current_user.jurisdiction
-    redirect_to(root_url) && return unless (cur_jur.descendant_ids + [cur_jur.id]).include? user.jurisdiction.id
+    users.each do |user|
+      #TODO: do we want this to just move on here or should we initially check that all pass this validation?
```

Comment:
@mayerm94 @timwongj I'd like your two cents here. This check was already happening on this method when it only handled one user at a time. Now that it can handle resetting for multiple at a time, we have a choice to either check all users pass this validation first, or just skip the ones that don't. Right now it's skipping the ones that don't. Thoughts?

Reply:
I'm thinking that we can just skip the ones that don't but display some kind of alert/warning to let the user know that those were skipped. This sounds like more of a design question though. @holmesie thoughts?

Reply:
I think it should be all or nothing. Assuming that the frontend is operating as expected, the user should never have access to patients who do not meet this criteria. In the case that a request is attempting to access resources outside of their allowed scope I think that we should just throw the request away.

Reply:
I'm leaning more towards that as well @mayerm94, for exactly the reason you said: this should never happen, so if it does, that's worrying and I don't want to make any changes from it at all.

Reply:
Done and tested in controller tests.